WooCommerce/Classes/ViewRelated/Products/ProductsViewController.swift

```diff
@@ -499,23 +513,50 @@ private extension ProductsViewController {
                                                  sortOrder: sortOrder)
     }
 
+    /// Configure resultController.
+    /// Assign closures and start performBatch
+    ///
     func configureResultsController(_ resultsController: ResultsController<StorageProduct>, onReload: @escaping () -> Void) {
+        setClosuresToResultController(resultsController, onReload: onReload)
+
+        do {
+            try resultsController.performFetch()
+        } catch {
+            ServiceLocator.crashLogging.logError(error)
+        }
+
+        tableView.reloadData()
+    }
+
+    /// Set closure  to methods `onDidChangeContent` and `onDidResetContent
+    ///
+    func setClosuresToResultController(_ resultsController: ResultsController<StorageProduct>, onReload: @escaping () -> Void) {
         resultsController.onDidChangeContent = {
             onReload()
         }
 
         resultsController.onDidResetContent = {
             onReload()
         }
+    }
 
-        do {
-            try resultsController.performFetch()
-        } catch {
-            ServiceLocator.crashLogging.logError(error)
-        }
-
+    /// Manages view components and reload tableview
+    ///
+    func reloadTableAndView() {
+        showOrHideToolBar()
+        removeOverlayIfControllerHasResults()
         tableView.reloadData()
     }
+
+    /// Try to remove Overlay if resultController has products
+    ///
+    func removeOverlayIfControllerHasResults() {
```

Comment:
I think we should also update the method name, as right now it solely implies that the overlay will be removed but not displayed.

Reply:
Totally forgot! Refactored!